File: backend/app/schemas/rut.py
from pydantic import BaseModel, Field, constr
from typing import Optional, Literal

class RutRequest(BaseModel):
    cedula: str = Field(
        ..., 
        description="Número de cédula colombiana a consultar",
        pattern=r"^\d{6,10}$",
        json_schema_extra={"example": "16728423"}
    )

class RutData(BaseModel):
    cedula_consultada: str
    contenido_confirmado: str

class RutResponse(BaseModel):
    status: Literal["success", "failed"]
    rut_exists: Optional[bool] = None
    data: Optional[RutData] = None
    message: str
    duration_ms: int

File: backend/src/schemas/rut.ts
import { z } from 'zod'

export const RutRequestSchema = z.object({
  cedula: z.string().regex(/^\d{6,10}$/, 'Cédula debe ser numérica de 6-10 dígitos'),
})

export const RutResponseSchema = z.object({
  status: z.enum(['success', 'failed']),
  rut_exists: z.boolean().nullable(),
  data: z.object({
    cedula_consultada: z.string(),
    contenido_confirmado: z.string(),
  }).nullable(),
  message: z.string(),
  duration_ms: z.number(),
})

export type RutRequest = z.infer<typeof RutRequestSchema>
export type RutResponse = z.infer<typeof RutResponseSchema>
